### ISSUE TITLE
feat: Implement daily quests 🚧 

### ISSUE THREAD

**Issue by apomalyn** [RoadTripMoustache/kana_to_kanji#338]
As a user, I want a daily goal to get me motivated every day

## Requirements

Implement daily quests system that:
- generate 4 quests every day, see below for more information about how to generate them
- can update the current status of a quest
- those quests should be synced with the API to allow future cross-platform move
- have a widget to display the quest

### Quests 

Type of quests:
- questions based:
  - Do X _Question format_ exercises, with X between 4 and 12
- Resources based:
  - Learn X new _hiragana | katakana | kanji | vocabulary (words)_, with X between 2 and 10
  - Do X lessons, with X between 1 and 4 **not available yet**
- Quiz based:
  - Do X auto-quizzes, with X between 3 and 8
  - Create a custom quiz, which can only appear if the user hasn't created a custom quiz yet
  - Do X quizzes, with X between 2 and 8

**How to generate them?**

1. It should not have 2 identical quests on the same day even if the difficulties is different
2. There should be increasing difficulties between the generated quests, it should have 1 easy quest, 2 medium and 1 hard quest or 1 medium and 2 hard quests.

## Design

[Figma link ](https://www.figma.com/design/ynARb6x2tOibdeCCOq7PgH/App?node-id=395-6581&t=1D1tiyiUlmCf6Jz4-4) to the section

<details>
    <summary><i>Today's milestone</i> section</summary> 
    

![image](https://github.com/user-attachments/assets/5dcb41ee-bfae-465c-ac62-600f662abee8)

</details>

## Dependencies

Child of #334 

Parent of:

- #issue_number

**Comment by Adaendra** [comment]
@apomalyn 

# Requirements
- May be add "sync the achievements" ?

# Quests
- Resources based
    - 10 new words or kanji can be a lot for a daily quest no ? Can we split between kanas and kanji/vocabulary it can be better as we can have a lower max for kanji/vocabulary
- Quiz based
    - "Do X quizzes" means which quizzes ? Any of them ? or specific for the custom quizzes ? If it's only for custom quizzes, we need to be more precise

## How to generate them?
- Which kind of quests are "easy", "medium" or "hard" ?

---

# Other questions 
- Is there any reward for completing the quizzes ? If not, most of the users won't be interested in. May be collecting the "in game money" to be able to temporarily unlock some premium features
- Other quests ideas
    - Complete X minigames (or Y of specific game)
    - Read an article / story
    - Add a friend (??)
    - Learn X minutes
    - Do X kanji drawing exercice

**Comment by apomalyn** [comment]
> May be add "sync the achievements" ?

What do you mean? Sync the quests with the achievements?

> Resources based
> 10 new words or kanji can be a lot for a daily quest no ? Can we split between kanas and 
> kanji/vocabulary it can be better as we can have a lower max for kanji/vocabulary

We can link kanji and vocabulary together and link the kanas also. The 10 would be a _hard_ quest.

> "Do X quizzes" means which quizzes ? Any of them ? or specific for the custom quizzes ? If it's only for custom quizzes, we need to be more precise

It's any quizzes

> Which kind of quests are "easy", "medium" or "hard" ?

The difference between them is the goal to achieve. My guess is 30% of the max value. For example, if the goal is between 2 and 10: 
- Easy: 2 and 3
- Medium: 3 and 6
- Hard: 6 and 10

What do you think?

> Is there any reward for completing the quizzes ? If not, most of the users won't be interested in. May be collecting the "in game money" to be able to temporarily unlock some premium features

For now, there is no reward, we talked about having in game money (and it's on the Idees en vrac) but nothing yet. Could be planned for v1, or we could add the in game money with no uses for now (hidden from the user until we have can use it)

> Complete X minigames (or Y of specific game)
Read an article / story
Learn X minutes

Added.

> Add a friend (??)

Friend system added to the Idee en vrac board.

> Do X kanji drawing exercise

Already there: _Do X Question format exercises_

**Comment by Adaendra** [comment]
> What do you mean? Sync the quests with the achievements?
Yes. As we have achievements, we must update them if some are related to quests.

For everything else : 👍

**Comment by apomalyn** [comment]
> Yes. As we have achievements, we must update them if some are related to quests.

We have the events observer for that (#155) but I added that when a quest is confirmed an event (`quest.finished`) should be fired
